crates/bevy_gizmos/src/primitives/dim2.rs

```diff
@@ -637,14 +634,16 @@ where
             return;
         }
 
-        let direction = self.direction * self.half_length;
-        let start = self.isometry * (-direction);
-        let end = self.isometry * direction;
+        let segment = Segment2d::new(self.point1, self.point2)
+            .translated(self.isometry.translation)
+            .rotated_around_center(self.isometry.rotation);
```

Comment:
Picture a collider with a segment shape that is offset relative to the transform of the entity. The segment should stay at the same position in the local space of the collider even when that entity rotates. In other words, the segment should rotate around the center of that collider, not around its own center.

This is currently translating the segment to the origin and then applying the isometry, rather than applying the isometry directly to the vertices. I think this should instead be `rotated` rather than rotating around the center of the segment. Also, rotation should be done before translation for [isometries](https://docs.rs/bevy_math/latest/bevy_math/struct.Isometry2d.html).
```suggestion
        let segment = Segment2d::new(self.point1, self.point2)
            .rotated(self.isometry.rotation)
            .translated(self.isometry.translation);
```